src/ecdsa/keygen.rs

```diff
@@ -790,16 +819,13 @@ impl State<KeyGeneratorTraits> for Phase3 {
 
         shares.values_mut().for_each(|x| x.zeroize());
 
-        let public_key = self
-            .pubkey_map
-            .iter()
-            .fold(None, |acc, (_party, key)| {
-                Some(acc.map_or_else(
-                    || Secp256k1Point::from_coor(&key.x_coor().unwrap(), &key.y_coor().unwrap()),
-                    |sum| sum + key,
-                ))
-            })
-            .expect("Points must add up");
+        let (public_key, pk_verification_errors) = try_computing_public_key(&self.pubkey_map);
```

Comment:
With `try_computing_public_key2` this will look like
```rust
let public_key = match try_computing_public_key2(&self.pubkey_map) {
            Ok(pk) => pk,
            Err(errs) => {
                errors.extend(errs);
                // Impossibility to reconstruct public key is a terminal error.
                return Transition::FinalState(Err(ErrorState::new(errors)));
            }
        };
```

Reply:
Unfortunately this poses early exit.

Reply:
I see, we probably want to collect this error too `self.secret_key_loader.get_paillier_secret()` and then exit early.